suave/core/engine.go

```diff
@@ -185,82 +163,74 @@ func (e *ConfidentialStoreEngine) FetchBidsByProtocolAndBlock(blockNumber uint64
 	return e.backend.FetchBidsByProtocolAndBlock(blockNumber, namespace)
 }
 
-func (e *ConfidentialStoreEngine) Store(bidId BidId, sourceTx *types.Transaction, caller common.Address, key string, value []byte) (Bid, error) {
+func (e *ConfidentialStoreEngine) Retrieve(bidId BidId, caller common.Address, key string) ([]byte, error) {
 	bid, err := e.backend.FetchBidById(bidId)
 	if err != nil {
-		return Bid{}, fmt.Errorf("confidential engine: could not fetch bid %x while storing: %w", bidId, err)
+		return []byte{}, fmt.Errorf("confidential engine: could not fetch bid %x while retrieving: %w", bidId, err)
 	}
 
 	if !slices.Contains(bid.AllowedPeekers, caller) {
-		return Bid{}, fmt.Errorf("confidential engine: %x not allowed to store %s on %x", caller, key, bidId)
+		return []byte{}, fmt.Errorf("confidential engine: %x not allowed to retrieve %s on %x", caller, key, bidId)
 	}
 
-	msg := DAMessage{
-		Bid:       bid,
-		SourceTx:  sourceTx,
-		Caller:    caller,
-		Key:       key,
-		Value:     value,
-		StoreUUID: e.storeUUID,
-	}
+	return e.backend.Retrieve(bid, caller, key)
+}
 
-	msgBytes, err := SerializeMessageForSigning(msg)
-	if err != nil {
-		return Bid{}, fmt.Errorf("confidential engine: could not hash message for signing: %w", err)
+func (e *ConfidentialStoreEngine) Finalize(tx *types.Transaction, newBids map[BidId]Bid, stores []StoreWrite) error {
```

Comment:
Out of the scope of the review, but would this be possible?
```suggestion
type ConfStoreUpdate struct {
   Bid Bid
   Writes []ConfStoreWrite
}
type ConfStoreWrite struct {
   Key []byte
   ...
}
func (e *ConfidentialStoreEngine) Finalize(tx *types.Transaction, updates []ConfStoreUpdate) error {
```

Reply:
Hm, no - the individual writes are not tied to any one bid. They each have their own bid